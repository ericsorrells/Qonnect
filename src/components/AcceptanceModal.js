// ========================================================================================
import React from 'react';
import Modal from 'react-modal';
// ========================================================================================
import ShowEventDisplay from '../components/ShowEventDisplay';
// ========================================================================================

class AcceptanceModal extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      acceptanceNote: ''
    }

    this.onNoteInput = this.onNoteInput.bind(this);
    this.onFormSubmit = this.onFormSubmit.bind(this);
  }

  onNoteInput(e) {
    event.preventDefault();
    this.setState({ acceptanceNote: e.target.value })
  }

  onFormSubmit(event){
    event.preventDefault();
    this.props.userAcceptEvent({
      acceptanceNote: this.state.acceptanceNote
    })
  }

  render(){
    return (
      <Modal
        isOpen={this.props.modalOpen}
        onRequestClose={this.props.closeModal}
        // style={customStyles}
        contentLabel="Example Modal"
      >
        <h2>Pleases Confirm You Wish To Accept This Event:</h2>
        <ShowEventDisplay event={this.props.event} />
        <form onSubmit={this.onFormSubmit}>
          <label>Acceptance Note:</label>
          <textarea
            onChange={this.onNoteInput}
            value={this.state.acceptanceNote}
          />
          <button type='submit'>Yes, I Accept!</button>
        </form>
      </Modal>
    )
  }
}

export default AcceptanceModal;