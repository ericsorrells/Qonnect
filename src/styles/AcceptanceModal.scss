.Modal {
	padding: 0;
	position: absolute;
	top: 40px;
	left: 40px;
	right: 40px;
	bottom: 40px;
	border: 1px solid rgb(204, 204, 204);
	background: rgb(255, 255, 255);
	overflow: auto;
	border-radius: 5px;
	outline: none;
	width: 50%;
	margin-left: auto;
	margin-right: auto;
}

.modal__container {
	padding-left: 5%;
	padding-right: 5%;
	margin: 20px 0;
}

.modal__title {
	background-color: $main-color;
	color: white;
	padding: 20px 5%;
	margin: 0;
}

.modal__label {
	font-weight: bold;
}
.modal__textarea {
	width: 75%;
  height: 100px;
  border-radius: 5px;
}

.modal__button {
	width: 75%;
	margin-top: 20px;
}


