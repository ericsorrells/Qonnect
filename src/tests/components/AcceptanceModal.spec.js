// ========================================================================================
import React from 'react';
import { shallow } from 'enzyme';
import 'jest-enzyme';
// ========================================================================================
import AcceptanceModal from '../../components/AcceptanceModal';
// ========================================================================================
import { basicEvent } from '../stubs/event';
import { basicProfile } from '../stubs/profile'
// ========================================================================================

describe('<AcceptanceModal />', () => {
  let component, userAcceptEvent;
  beforeEach(() => {
    userAcceptEvent = jest.fn();
    component = shallow(
      <AcceptanceModal
        modalOpen={true}
        closeModal={jest.fn()}
        userAcceptEvent={userAcceptEvent}
        event={basicEvent}
      />
    )
  });

  describe('rendering', () => {
    it('renders initial state correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('handles data entry correctly', () => {
    it('has the correct inital state values', () => {
      expect(component.state('acceptanceNote')).toBe('')
    });

    it('allow user to enter an acceptance message', () => {
      const value = 'Id like to go to this event';
      component.find('.modal__textarea').simulate('change', { target: { value } });
      expect(component.state('acceptanceNote')).toBe(value)
    });

    it('allows user to accept an event', () => {
      const value = 'Id like to go to this event';
      component.find('.modal__textarea').simulate('change', { target: { value } });

      component.find('form').simulate('submit', {
        preventDefault: () => { }
      });

      expect(userAcceptEvent).toHaveBeenLastCalledWith({
        acceptanceNote: value
      });
    });
  });
});
