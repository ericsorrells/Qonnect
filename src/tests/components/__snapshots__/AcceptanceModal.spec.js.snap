// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AcceptanceModal /> rendering renders initial state correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="Modal"
  closeTimeoutMS={0}
  contentLabel="Example Modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title modal__container"
  >
    Please Confirm You Wish To Accept This Event:
  </h3>
  <ShowEventDisplay
    event={
      Object {
        "category": "sports",
        "date": 345600000,
        "description": "Lorem ipsum text",
        "eventName": "Cool Event Title",
        "id": "-LDTTYL9PSqqroO6JLYh",
        "imageUrl": "https://image.com",
        "interestedUsers": "none",
        "location": "123 Elm St.",
        "selectedUser": "none",
        "time": "02:13",
        "uid": "abc123",
        "uninterestedUsers": "none",
        "userId": "abc123",
        "userName": "Rachel Deckard",
      }
    }
  />
  <form
    className="modal__container"
    onSubmit={[Function]}
  >
    <label
      className="modal__label"
    >
      Acceptance Note:
    </label>
    <textarea
      className="modal__textarea"
      onChange={[Function]}
      value=""
    />
    <br />
    <button
      className="button__purple modal__button"
      type="submit"
    >
      Yes, I Accept!
    </button>
  </form>
</Modal>
`;
